iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/performer/impl/FastCompactionPerformer.java

```diff
@@ -239,6 +240,8 @@ private void compactNonAlignedSeries(
     Map<String, Map<TsFileResource, Pair<Long, Long>>> timeseriesMetadataOffsetMap =
         deviceIterator.getTimeseriesMetadataOffsetOfCurrentDevice();
 
+    Map<String, TSDataType> measurementDataTypeMap = deviceIterator.getDataTypeOfCurrentDevice();
```

Comment:
Can these two methods be merged into one? Otherwise, the metadata is traversed twice, which may affect the efficiency.
Like 
```
Map<String, CompactionSeriesContext> timeseriesMetadataOffsetMap =
        deviceIterator.getCompactionSeriesContextOfCurrentDevice();
class CompactionSeriesContext {
    Map<TsFileResource, Pair<Long, Long>> fileTimeseriesMetdataOffsetMap;
    TSDataType finalType;
}
```